sc/memiavl/tree.go

```diff
@@ -116,6 +115,8 @@ func (t *Tree) ApplyChangeSet(changeSet iavl.ChangeSet) {
 }
 
 func (t *Tree) Set(key, value []byte) {
+	t.mtx.RLock()
```

Comment:
It should be a Lock?

Reply:
For us, we don't really need a write lock, because set, remove, get and iterator can be run in parallel as before, so they should all be read lock, and we only need a write lock for close and replaceWith

Reply:
isn't it modifies the `root` field, and the `MemNode`s are modified in place as well.

Reply:
isn't it modifies the `root` node, and also the `MemNode`s

Reply:
Ah you're correct, we do need to make this a write lock